#150776 breaks building utsushi package

### Describe the bug

The change introduced in #150776 breaks building utsushi package (on my system)

### Steps To Reproduce
Steps to reproduce the behavior:

Add package like this:

```nix
{
    hardware.sane.enable = true;
    hardware.sane.extraBackends = with pkgs; [
      epkowa
      utsushi
    ];
    services.udev.packages = [ pkgs.utsushi ];
}
```

Results in an error during the build:

<details><summary>Output of `nix log /nix/store/dn3nfllgnfh8sn90vxds9vh7j756s541-imagescan-3.65.0.drv`</summary>

```
 @nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/3nj9xsc03aqdrfkgbn448y3x2j91lmag-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/ndpg46zh2wf7hxp820rwjv7w4ib0h83h-iscan-3.63.0-autoconf-2.70.patch?id=4fe8a9e6c60f9163cadad830ba4935c069c67b10
patching file configure.ac
applying patch /nix/store/8bpsxnwlghzq8ww11c6lyd33wmnjrmi7-iscan-3.61.0-imagemagick-7.patch?id=985c92af4730d864e86fa87746185b0246e9db93
patching file filters/doc-locate.cpp
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
Copying file upstream/aclocal/codeset.m4
Copying file upstream/aclocal/extern-inline.m4
Copying file upstream/aclocal/fcntl-o.m4
Copying file upstream/aclocal/glibc2.m4
Copying file upstream/aclocal/glibc21.m4
Copying file upstream/aclocal/intdiv0.m4
Copying file upstream/aclocal/intl.m4
Copying file upstream/aclocal/intldir.m4
Copying file upstream/aclocal/intmax.m4
Copying file upstream/aclocal/inttypes-pri.m4
Copying file upstream/aclocal/inttypes_h.m4
Copying file upstream/aclocal/lcmessage.m4
Copying file upstream/aclocal/lock.m4
Copying file upstream/aclocal/longlong.m4
Copying file upstream/aclocal/printf-posix.m4
Copying file upstream/aclocal/size_max.m4
Copying file upstream/aclocal/stdint_h.m4
Copying file upstream/aclocal/threadlib.m4
Copying file upstream/aclocal/uintmax_t.m4
Copying file upstream/aclocal/visibility.m4
Copying file upstream/aclocal/wchar_t.m4
Copying file upstream/aclocal/wint_t.m4
Copying file upstream/aclocal/xsize.m4
Copying file po/Makevars.template
autoreconf: running: aclocal --force -I upstream/aclocal
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force --ltdl
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'upstream'.
libtoolize: copying file 'upstream/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'upstream/aclocal'.
libtoolize: copying file 'upstream/aclocal/libtool.m4'
libtoolize: copying file 'upstream/aclocal/ltargz.m4'
libtoolize: copying file 'upstream/aclocal/ltdl.m4'
libtoolize: copying file 'upstream/aclocal/ltoptions.m4'
libtoolize: copying file 'upstream/aclocal/ltsugar.m4'
libtoolize: copying file 'upstream/aclocal/ltversion.m4'
libtoolize: copying file 'upstream/aclocal/lt~obsolete.m4'
libtoolize: putting libltdl files in LT_CONFIG_LTDL_DIR, 'upstream/ltdl'.
libtoolize: copying file 'upstream/ltdl/COPYING.LIB'
libtoolize: creating file 'upstream/ltdl/Makefile.am'
libtoolize: copying file 'upstream/ltdl/README'
libtoolize: creating file 'upstream/ltdl/Makefile.in'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__alloc.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__argz_.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__dirent.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__glibc.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__private.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt__strl.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt_dlloader.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt_error.h'
libtoolize: copying file 'upstream/ltdl/libltdl/lt_system.h'
libtoolize: copying file 'upstream/ltdl/libltdl/slist.h'
libtoolize: copying file 'upstream/ltdl/loaders/dld_link.c'
libtoolize: copying file 'upstream/ltdl/loaders/dlopen.c'
libtoolize: copying file 'upstream/ltdl/loaders/dyld.c'
libtoolize: copying file 'upstream/ltdl/loaders/load_add_on.c'
libtoolize: copying file 'upstream/ltdl/loaders/loadlibrary.c'
libtoolize: copying file 'upstream/ltdl/loaders/preopen.c'
libtoolize: copying file 'upstream/ltdl/loaders/shl_load.c'
libtoolize: copying file 'upstream/ltdl/lt__alloc.c'
libtoolize: copying file 'upstream/ltdl/lt__argz.c'
libtoolize: copying file 'upstream/ltdl/lt__dirent.c'
libtoolize: copying file 'upstream/ltdl/lt__strl.c'
libtoolize: copying file 'upstream/ltdl/lt_dlloader.c'
libtoolize: copying file 'upstream/ltdl/lt_error.c'
libtoolize: copying file 'upstream/ltdl/ltdl.c'
libtoolize: copying file 'upstream/ltdl/ltdl.h'
libtoolize: copying file 'upstream/ltdl/slist.c'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I upstream/aclocal
autoreconf: running: /nix/store/khilbv2w0ixj55clcpsjzpaq0l1bhh2j-autoconf-2.71/bin/autoconf --force
autoreconf: running: /nix/store/khilbv2w0ixj55clcpsjzpaq0l1bhh2j-autoconf-2.71/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:208: installing 'upstream/compile'
configure.ac:204: installing 'upstream/missing'
connexions/Makefile.am: installing 'upstream/depcomp'
autoreconf: Leaving directory '.'
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
fixing libtool script ./upstream/ltmain.sh
fixing libtool script ./upstream/boost/tools/build/v2/engine/boehm_gc/ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0 --with-boost-libdir=/nix/store/ryqkcl29gbspipvl8gdan8gkrq356xyk-boost-1.77.0/lib --with-sane-confdir=/nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/sane.d --with-udev-confdir=/nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/udev --with-gtkmm --with-jpeg --with-magick --with-magick-pp --with-sane --with-tiff
checking for a BSD-compatible install... /nix/store/abc8hz0v47jzsf048j1qx7m94bxfhpl7-coreutils-9.0/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/abc8hz0v47jzsf048j1qx7m94bxfhpl7-coreutils-9.0/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1000' is supported by ustar format... yes
checking whether GID '100' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking for gcc option to support OpenMP... -fopenmp
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... none
checking standard compliance level of g++... 201402L
checking the archiver (ar) interface... ar
checking pkg-config is at least version 0.9.0... yes
checking for a sed that does not truncate output... /nix/store/aw3vgxn4fdrjqgmc94fr582hjhr2z3p3-gnused-4.8/bin/sed
checking whether NLS is requested... yes
checking for msgfmt... /nix/store/shv8hvhagfzpw7sm9a0qaxxgdvvngnzp-gettext-0.21/bin/msgfmt
checking for gmsgfmt... /nix/store/shv8hvhagfzpw7sm9a0qaxxgdvvngnzp-gettext-0.21/bin/msgfmt
checking for xgettext... /nix/store/shv8hvhagfzpw7sm9a0qaxxgdvvngnzp-gettext-0.21/bin/xgettext
checking for msgmerge... /nix/store/shv8hvhagfzpw7sm9a0qaxxgdvvngnzp-gettext-0.21/bin/msgmerge
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for shared library run path origin... done
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /nix/store/rvdb7057bmg3rasbqq8lzl10dffy3p31-gnugrep-3.7/bin/grep
checking for egrep... /nix/store/rvdb7057bmg3rasbqq8lzl10dffy3p31-gnugrep-3.7/bin/grep -E
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /nix/store/aw3vgxn4fdrjqgmc94fr582hjhr2z3p3-gnused-4.8/bin/sed
checking for fgrep... /nix/store/rvdb7057bmg3rasbqq8lzl10dffy3p31-gnugrep-3.7/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/abc8hz0v47jzsf048j1qx7m94bxfhpl7-coreutils-9.0/bin/dd
checking how to truncate binary pipes... /nix/store/abc8hz0v47jzsf048j1qx7m94bxfhpl7-coreutils-9.0/bin/dd bs=4096 count=1
./configure: line 11993: /usr/bin/file: No such file or directory
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for shl_load... no
checking for shl_load in -ldld... no
checking for dlopen... no
checking for dlopen in -ldl... yes
checking whether a program can dlopen itself... yes
checking whether a statically linked program can dlopen itself... yes
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld
checking if the linker (ld) is GNU ld... yes
checking whether the g++ linker (ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for gcc options needed to detect all undeclared functions... none needed
checking what extension is used for runtime loadable modules... .so
checking what variable specifies run-time module search path... LD_LIBRARY_PATH
checking for the default library search path... /lib /usr/lib
checking for library containing dlopen... -ldl
checking for dlerror... yes
checking for shl_load... (cached) no
checking for shl_load in -ldld... (cached) no
checking for dld_link in -ldld... no
checking for _ prefix in compiled symbols... no
checking whether deplibs are loaded by dlopen... yes
checking for argz.h... yes
checking for error_t... yes
checking for argz_add... yes
checking for argz_append... yes
checking for argz_count... yes
checking for argz_create_sep... yes
checking for argz_insert... yes
checking for argz_next... yes
checking for argz_stringify... yes
checking if argz actually works... yes
checking whether libtool supports -dlopen/-dlpreopen... yes
checking for ltdl.h... yes
checking whether lt_dlinterface_register is declared... yes
checking for lt_dladvise_preload in -lltdl... yes
checking where to find libltdl headers... 
checking where to find libltdl library... -lltdl
checking for unistd.h... (cached) yes
checking for dl.h... no
checking for sys/dl.h... no
checking for dld.h... no
checking for mach-o/dyld.h... no
checking for dirent.h... yes
checking for closedir... yes
checking for opendir... yes
checking for readdir... yes
checking for strlcat... no
checking for strlcpy... no
checking for boostlib >= 1.49.0... yes
checking whether the Boost::System library is available... yes
checking for exit in -lboost_system... yes
checking whether the Boost::Filesystem library is available... yes
checking for exit in -lboost_filesystem... yes
checking whether the Boost::IOStreams library is available... yes
checking for exit in -lboost_iostreams... yes
checking whether the Boost::Program_Options library is available... yes
checking for exit in -lboost_program_options... yes
checking whether the Boost::Unit_Test_Framework library is available... yes
checking for doxygen... no
checking for perl... no
checking for dot... no
checking for pdflatex... no
checking for LIBUDEV... yes
checking for LIBUSB... yes
checking for LIBMAGICK_PP... no
checking for LIBMAGICK_PP... yes
checking for gm... no
checking for convert... convert
checking for LIBGTKMM... yes
checking for sane/sane.h... yes
checking SANE version... 1.0.32
checking where SANE backends are installed... /nix/store/9rz898v8nrnyz2z2als2sbfpvr3gywk4-sane-backends-1.0.32/lib/sane
checking where SANE configuration is kept... /nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/sane.d
checking for /nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/sane.d/dll.conf... yes
checking what udev configuration directory to use... /nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/udev
checking for /nix/store/5k24i0bqgcaylfyvq34amgjh7nvj4nmn-imagescan-3.65.0/etc/udev/rules.d... yes
checking for uncrustify... no
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for C/C++ restrict keyword... __restrict__
checking size of unsigned long long... 8
checking for __attribute__... yes
checking for nanosleep... yes
checking for poll... yes
checking for sleep... yes
checking for usleep... yes
checking for sqrt in -lm... yes
checking for jpeg_start_compress in -ljpeg... yes
checking for TIFFOpen in -ltiff... yes
checking for magic_open in -lmagic... no
checking for libusb_error_name in -lusb-1.0... yes
checking for libusb_get_port_number in -lusb-1.0... yes
checking for sigaction... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating connexions/Makefile
config.status: creating doc/Makefile
config.status: creating doc/tests/Makefile
config.status: creating drivers/Makefile
config.status: creating drivers/esci/Makefile
config.status: creating drivers/esci/tests/Makefile
config.status: creating filters/Makefile
config.status: creating filters/tests/Makefile
config.status: creating gtkmm/Makefile
config.status: creating gtkmm/about.xml
config.status: creating lib/Makefile
config.status: creating lib/tests/Makefile
config.status: creating sane/Makefile
config.status: creating sane/tests/Makefile
config.status: creating src/Makefile
config.status: creating src/tests/Makefile
config.status: creating tests/Makefile
config.status: creating po/Makefile.in
config.status: creating upstream/ltdl/Makefile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing po-directories commands
config.status: creating po/POTFILES
config.status: creating po/Makefile
config.status: executing libtool commands
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j8 -l8 SHELL=/nix/store/iqprjr5k5385bhf1dzj07zwd5p43py1n-bash-5.1-p12/bin/bash
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in .
make[2]: Entering directory '/build/source'
make[2]: Leaving directory '/build/source'
Making all in lib
make[2]: Entering directory '/build/source/lib'
Making all in .
make[3]: Entering directory '/build/source/lib'
  CXX      connexion.lo
  CXX      device-info.lo
  CXX      exception.lo
  CXX      log.lo
  CXX      monitor.lo
  CXX      run-time.lo
  CXX      scanner.lo
  CXX      octet.lo
  CXX      context.lo
  CXX      iobase.lo
  CXX      device.lo
  CXX      filter.lo
  CXX      buffer.lo
[01m[Kmonitor.cpp:[m[K In member function '[01m[Kstd::string utsushi::monitor::default_device() const[m[K':
[01m[Kmonitor.cpp:79:56:[m[K [01;31m[Kerror: [m[K'[01m[K_1[m[K' was not declared in this scope
   79 |                   bind (&scanner::info::is_driver_set, [01;31m[K_1[m[K));
      |                                                        [01;31m[K^~[m[K
[01m[Kmonitor.cpp:79:56:[m[K [01;36m[Knote: [m[Ksuggested alternatives:
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/system/detail/error_category.hpp:17[m[K,
                 from [01m[Kinclude/boost/system/error_category.hpp:10[m[K,
                 from [01m[Kinclude/boost/filesystem/path_traits.hpp:14[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:21[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[K/nix/store/v819nrv8d33ns36gm4v9vqydq1v95axi-gcc-10.3.0/include/c++/10.3.0/functional:221:34:[m[K [01;36m[Knote: [m[K  '[01m[Kstd::placeholders::_1[m[K'
  221 |     extern const _Placeholder<1> [01;36m[K_1[m[K;
      |                                  [01;36m[K^~[m[K
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/bind/bind.hpp:2336[m[K,
                 from [01m[Kinclude/boost/signals2/slot.hpp:15[m[K,
                 from [01m[Kinclude/boost/signals2/connection.hpp:25[m[K,
                 from [01m[K../utsushi/signal.hpp:28[m[K,
                 from [01m[K../utsushi/device.hpp:29[m[K,
                 from [01m[K../utsushi/scanner.hpp:27[m[K,
                 from [01m[K../utsushi/monitor.hpp:31[m[K,
                 from [01m[Kmonitor.cpp:41[m[K:
[01m[Kinclude/boost/bind/placeholders.hpp:58:38:[m[K [01;36m[Knote: [m[K  '[01m[Kboost::placeholders::_1[m[K'
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
[01m[Kmonitor.cpp:[m[K In function '[01m[Kbool utsushi::is_all_device_connected(const container_type&, const id_list&)[m[K':
[01m[Kmonitor.cpp:524:27:[m[K [01;31m[Kerror: [m[K'[01m[K_1[m[K' was not declared in this scope
  524 |                           [01;31m[K_1[m[K, id->first, id->second));
      |                           [01;31m[K^~[m[K
[01m[Kmonitor.cpp:524:27:[m[K [01;36m[Knote: [m[Ksuggested alternatives:
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/system/detail/error_category.hpp:17[m[K,
                 from [01m[Kinclude/boost/system/error_category.hpp:10[m[K,
                 from [01m[Kinclude/boost/filesystem/path_traits.hpp:14[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:21[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[K/nix/store/v819nrv8d33ns36gm4v9vqydq1v95axi-gcc-10.3.0/include/c++/10.3.0/functional:221:34:[m[K [01;36m[Knote: [m[K  '[01m[Kstd::placeholders::_1[m[K'
  221 |     extern const _Placeholder<1> [01;36m[K_1[m[K;
      |                                  [01;36m[K^~[m[K
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/bind/bind.hpp:2336[m[K,
                 from [01m[Kinclude/boost/signals2/slot.hpp:15[m[K,
                 from [01m[Kinclude/boost/signals2/connection.hpp:25[m[K,
                 from [01m[K../utsushi/signal.hpp:28[m[K,
                 from [01m[K../utsushi/device.hpp:29[m[K,
                 from [01m[K../utsushi/scanner.hpp:27[m[K,
                 from [01m[K../utsushi/monitor.hpp:31[m[K,
                 from [01m[Kmonitor.cpp:41[m[K:
[01m[Kinclude/boost/bind/placeholders.hpp:58:38:[m[K [01;36m[Knote: [m[K  '[01m[Kboost::placeholders::_1[m[K'
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
[01m[Kmonitor.cpp:[m[K In function '[01m[Kvoid utsushi::configure_combo_device(utsushi::monitor::container_type&)[m[K':
[01m[Kmonitor.cpp:557:41:[m[K [01;31m[Kerror: [m[K'[01m[K_1[m[K' was not declared in this scope
  557 |                                         [01;31m[K_1[m[K, id->first, id->second));
      |                                         [01;31m[K^~[m[K
[01m[Kmonitor.cpp:557:41:[m[K [01;36m[Knote: [m[Ksuggested alternatives:
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/system/detail/error_category.hpp:17[m[K,
                 from [01m[Kinclude/boost/system/error_category.hpp:10[m[K,
                 from [01m[Kinclude/boost/filesystem/path_traits.hpp:14[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:21[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[K/nix/store/v819nrv8d33ns36gm4v9vqydq1v95axi-gcc-10.3.0/include/c++/10.3.0/functional:221:34:[m[K [01;36m[Knote: [m[K  '[01m[Kstd::placeholders::_1[m[K'
  221 |     extern const _Placeholder<1> [01;36m[K_1[m[K;
      |                                  [01;36m[K^~[m[K
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
In file included from [01m[Kinclude/boost/bind/bind.hpp:2336[m[K,
                 from [01m[Kinclude/boost/signals2/slot.hpp:15[m[K,
                 from [01m[Kinclude/boost/signals2/connection.hpp:25[m[K,
                 from [01m[K../utsushi/signal.hpp:28[m[K,
                 from [01m[K../utsushi/device.hpp:29[m[K,
                 from [01m[K../utsushi/scanner.hpp:27[m[K,
                 from [01m[K../utsushi/monitor.hpp:31[m[K,
                 from [01m[Kmonitor.cpp:41[m[K:
[01m[Kinclude/boost/bind/placeholders.hpp:58:38:[m[K [01;36m[Knote: [m[K  '[01m[Kboost::placeholders::_1[m[K'
In file included from [01m[Kinclude/boost/mpl/aux_/include_preprocessed.hpp:37[m[K,
                 from [01m[Kinclude/boost/mpl/placeholders.hpp:43[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_categories.hpp:16[m[K,
                 from [01m[Kinclude/boost/iterator/iterator_facade.hpp:13[m[K,
                 from [01m[Kinclude/boost/filesystem/path.hpp:22[m[K,
                 from [01m[Kinclude/boost/filesystem.hpp:16[m[K,
                 from [01m[Kmonitor.cpp:36[m[K:
[01m[Kinclude/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16:[m[K [01;36m[Knote: [m[K  '[01m[Kmpl_::_1[m[K'
make[3]: *** [Makefile:706: monitor.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/source/lib'
make[2]: *** [Makefile:722: all-recursive] Error 1
make[2]: Leaving directory '/build/source/lib'
make[1]: *** [Makefile:618: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:526: all] Error 2
****
```

</details>

When pinning the channel and installing the package to the previous revision b0249fdf998d782e1058b0cf3239091e59e393ef
the package doesn't break.

Please let me know if you need any more input

### Notify maintainers

@abbradar 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.5-zen1, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
```

--- a/pkgs/misc/drivers/utsushi/default.nix
+++ b/pkgs/misc/drivers/utsushi/default.nix
@@ -33,6 +33,10 @@ in stdenv.mkDerivation rec {
       url = "https://gitweb.gentoo.org/repo/gentoo.git/plain/media-gfx/iscan/files/iscan-3.61.0-imagemagick-7.patch?id=985c92af4730d864e86fa87746185b0246e9db93";
       sha256 = "sha256-dfdVMp3ZfclYeRxYjMIvl+ZdlLn9S+IwQ+OmlHW8318=";
     })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/archlinux/svntogit-community/b3046e0e78b95440f135fcadb19a9eb531729a58/trunk/boost-1.74.patch";
+      sha256 = "sha256-W8R1l7ZPcsfiIy1QBJvh0M8du0w1cnTg3PyAz65v4rE=";
+    })
   ];
 
   nativeBuildInputs = [

